exercises/matthew_cumber/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,144 @@
+#include "UniqueWord.h"
+
+#include <iostream>
+#include <iomanip>
+#include <algorithm>
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <ctype.h>
+
+/** Checks if a word already exists in the vector or words, then increments it's count if found,
+ * otherwise create a new word and add to vector
+ *
+ * @param uniqueWords The vector of existing unique words found
+ * @param word The word being tested
+ * @return No return value 
+ */
+void incrementCountOrAddNewWord(std::vector<UniqueWord> &uniqueWords, const std::string &word)
+{
+  // Attempt to find word in the vector uniqueWords
+  auto it = std::find_if(uniqueWords.begin(), uniqueWords.end(), [&word](const UniqueWord &obj){return obj.getWord() == word;});
+
+  // If not at end, then word has already been found in input file and is stored in the vector, so increment count
+  if(it != uniqueWords.end())
+    it->incrementCount();
+
+  // Else the word has not been found previously, so create a new instance and add it to the vector 
+  else 
+  {
+    auto newWord = UniqueWord(word, 1);
+    uniqueWords.push_back(newWord);
```

Comment:
In general we're trying to move towards using the function emplace_back rather than push_back because in some circumstances it gives better performance (when you're adding a temporary to a vector).
This isn't one of those cirucmstances but to avoid too much thought we're simply defaulting to using emplace_back in all cases so please could you swap to that